app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java

```diff
@@ -0,0 +1,38 @@
+package com.teamagam.gimelgimel.app.network.receivers;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+/**
+ * A receiver that is notified for any connectivity problem and changes.
+ * The receiver uses {@link NetworkAvailableListener} to notify other classes about any changes in the network.
+ */
+public class ConnectivityStatusReceiver extends BroadcastReceiver {
+
+    public static final String INTENT_NAME = "connectivityStatus";
+    public static final String NETWORK_AVAILABLE_EXTRA = "networkAvailable";
+
+    private NetworkAvailableListener mNetworkAvailableListener;
```

Comment:
If there's no point in instantiating this object without a listener, i'd say you should ask for one at the constructor